workflows/prognostic_c48_run/runtime/loop.py

```diff
@@ -449,15 +449,18 @@ def _apply_postphysics_to_dycore_state(self) -> Diagnostics:
             if self._postphysics_only_diagnostic_ml:
                 rename_diagnostics(diagnostics)
             else:
-                updated_state = add_tendency(self._state, tendency, dt=self._timestep)
-                updated_state[TOTAL_PRECIP] = precipitation_sum(
+                updated_state_from_tendency = add_tendency(
+                    self._state, tendency, dt=self._timestep
+                )
+                updated_state_from_tendency[TOTAL_PRECIP] = precipitation_sum(
                     self._state[TOTAL_PRECIP],
                     diagnostics[self._postphysics_stepper.net_moistening],
                     self._timestep,
                 )
-                diagnostics[TOTAL_PRECIP] = updated_state[TOTAL_PRECIP]
+                diagnostics[TOTAL_PRECIP] = updated_state_from_tendency[TOTAL_PRECIP]
+                self._state.update_mass_conserving(updated_state_from_tendency)
```

Comment:
Can you use one call to `self._state.update_mass_conserving`? The way it is currently written could imply it is important to split into two calls (e.g. it is relying on some state inside of `self._state`. This is harder to reason about.

Reply:
On second thought it _is_ currently important that there are two calls: when `total_precipitation` is predicted by prephysics ML, it's in both `updated_state_from_tendency` and in `self._state_updates`, and it's the one in the latter we actually want to use, which is why there's currently two calls with that going second. This is confusing and needs to get resolved, but I'm hesitant to overhaul this RN when it's the crux of the experiment we're about to run.